web/test/pages/onboarding/onboarding-owning.test.tsx

```diff
@@ -267,6 +268,32 @@ describe("onboarding - owning a business", () => {
     });
   });
 
+  it("removes welcome and adds welcome-up-and-running to visibleSidebarCards preferences on save", async () => {
+    const initialUserData = generateUserData({
+      profileData: generateProfileData({
+        businessPersona: "OWNING",
+        operatingPhase: "GUEST_MODE_OWNING",
+      }),
+      preferences: generatePreferences({
+        visibleSidebarCards: ["welcome"],
+      }),
+    });
+    const { page } = renderPage({ userData: initialUserData });
+    expect(page.getRadioButton("Business Status - Owning")).toBeChecked();
+    await page.visitStep(2);
+    page.clickNext();
+
+    await waitFor(() => {
+      expect(currentUserData()).toEqual({
+        ...initialUserData,
+        preferences: {
+          ...initialUserData.preferences,
+          visibleSidebarCards: ["welcome-up-and-running"],
```

Comment:
If this change is happening between when the user moves from step 1 to step 2, then I think you'll have to write another test / more code for this situation: a user selects OWNING on step 1, moves to step 2, then goes Back to step 1, and goes through Poppy flow instead. In that word, I think they would get the up-and-running card (but they SHOULD get the regular welcome card).

You could either add logic to "undo" the change if they are Poppy. Or you could just have the card-switch logic happen after submitting from the final step, instead of happening between step 1 and 2. That way there's no way to go back.

Reply:
If I'm not wrong, currently the code to add and remove of welcome cards happens in the else clause of the onSubmit function which fires on the final step.

Reply:
Got it, missed that. looks good